packages/core/src/new-api/internal/journal/utils/serialize-replacer.ts

```diff
@@ -0,0 +1,34 @@
+import { serializeError } from "serialize-error";
+
+/**
+ * When stringifying messages to the journal, this defines the replacer.
+ */
+export function serializeReplacer(_key: string, value: unknown) {
+  if (value instanceof Set) {
+    return Array.from(value).sort();
+  }
+
+  if (value instanceof Map) {
+    return Object.fromEntries(value);
+  }
+
+  if (typeof value === "bigint") {
+    return `${value.toString(10)}n`;
+  }
+
+  if (value instanceof Error) {
+    return serializeError(new Error(value.message));
+  }
```

Comment:
I think this is unsafe and we shouldn't try to serialize errors altogether. We are tricking ourselves into thinking that we are storing them, but we are loosing info (e.g. the stack trace).

Reply:
I have pulled this out as its own issue: https://github.com/NomicFoundation/ignition/issues/286